pytket/pytket/wasm/wasm.py

```diff
@@ -64,169 +72,181 @@ def __init__(self, filepath: str, check_file: bool = True, int_size: int = 32):
+            raise ValueError("init function should not have any parameter")
 
-            if self._functions["init"][0] != 0:
-                raise ValueError("init function should not have any parameter")
+        if self._functions["init"][1] != 0:
+            raise ValueError("init function should not have any results")
 
-            if self._functions["init"][1] != 0:
-                raise ValueError("init function should not have any results")
+        self.checked = True
+
+    @deprecated("Use public property checked instead.")
+    def _check_file(self) -> bool:
+        return self.checked
 
     def __str__(self) -> str:
-        """str representation of the wasm file"""
-        return self._wasmfileuid
+        """str representation of the wasm module"""
+        return self.module_uid
 
     def __repr__(self) -> str:
-        """str representation of the contents of the wasm file"""
-        if self._check_file:
-            result = f"Functions in wasm file with the uid {self._wasmfileuid}:\n"
-            for x in self._functions:
-                result += f"function '{x}' with "
-                result += f"{self._functions[x][0]} i{self._int_size} parameter(s)"
-                result += (
-                    f" and {self._functions[x][1]} i{self._int_size} return value(s)\n"
-                )
+        """str representation of the contents of the wasm file.
 
-            for x in self._unsupported_function:
-                result += (
-                    f"unsupported function with invalid "
-                    f"parameter or result type: '{x}' \n"
-                )
-
-            return result
-        else:
-            raise ValueError(
-                """the content of the wasm file can only be printed if the
-wasm file was checked"""
+        Will implicitly check the module if it has not been checked already.
+        """
+        result = f"Functions in wasm file with the uid {self.module_uid}:\n"
+        for x in self.functions:
+            result += f"function '{x}' with "
+            result += f"{self.functions[x][0]} i{self._int_size} parameter(s)"
+            result += f" and {self.functions[x][1]} i{self._int_size} return value(s)\n"
+
+        for x in self.unsupported_functions:
+            result += (
+                f"unsupported function with invalid "
+                f"parameter or result type: '{x}' \n"
             )
 
+        return result
+
     def bytecode(self) -> bytes:
         """The WASM content as bytecode"""
-        return self._wasm_file
+        return self._wasm_module
+
+    @cached_property
+    def bytecode_base64(self) -> bytes:
+        """The WASM content as base64 encoded bytecode."""
+        return base64.b64encode(self._wasm_module)
+
+    @deprecated("Use public property bytecode_base64 instead.")
+    def _wasm_file_encoded(self) -> bytes:
+        return self.bytecode_base64
+
+    @cached_property
+    def module_uid(self) -> str:
```

Comment:
Possibly this should just be called `uid` instead?

Reply:
Yes :+1: